File: src/main/java/org/example/commons/providers/EntityManagerProvider.java
package org.example.commons.providers;

import javax.enterprise.inject.Disposes;
import javax.enterprise.inject.Produces;
import javax.persistence.EntityManager;
import javax.persistence.Persistence;

public class EntityManagerProvider {

    @Produces
    private EntityManager getEntityManager() {
        return Persistence.createEntityManagerFactory("AppDB").createEntityManager();
    }

    private void destroy(@Disposes final EntityManager em){

    }
}

File: src/main/kotlin/org/example/commons/providers/EntityManagerProvider.kt
package org.example.commons.providers

import javax.enterprise.inject.Disposes
import javax.enterprise.inject.Produces
import javax.persistence.EntityManager
import javax.persistence.Persistence

object EntityManagerProvider {
    @get:Produces
    private val entityManager: EntityManager
        get() = Persistence.createEntityManagerFactory("AppDB").createEntityManager()

    private fun destroy(@Disposes em: EntityManager) {}
}